dotCMS/src/main/java/com/dotmarketing/portlets/workflows/business/WorkflowAPIImpl.java

```diff
@@ -1834,14 +1850,263 @@ private void updateTask(WorkflowProcessor processor) throws DotDataException{
 			task.setTitle(processor.getContentlet().getTitle());
 		}
 
-
 		if(processor.getWorkflowMessage() != null){
 			WorkflowComment comment = new WorkflowComment();
 			comment.setComment(processor.getWorkflowMessage());
 			comment.setWorkflowtaskId(task.getId());
 			saveComment(comment);
 		}
+	}
+
+	/**
+	 * Entry point that fires up the actions associated with the contentles. Expects a lucene query
+	 * that holds the logic to retrive a large selection of items performed on the UI.
+	 *
+	 * @param action {@link WorkflowAction}
+	 * @param user {@link User}
+	 * @param luceneQuery luceneQuery
+	 */
+	public Future<BulkActionsResultView> fireBulkActions(final WorkflowAction action,

```

Comment:
if you are calling a factory you need to use the CloseDBIfOpened to make sure you close your connection !important